its/src/test/java/com/sonar/it/scanner/msbuild/sonarqube/JreProvisioningTest.java

```diff
@@ -0,0 +1,122 @@
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package com.sonar.it.scanner.msbuild.sonarqube;
+
+import com.sonar.it.scanner.msbuild.utils.TestUtils;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.io.TempDir;
+
+import static com.sonar.it.scanner.msbuild.sonarqube.Tests.ORCHESTRATOR;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@ExtendWith(Tests.class)
+public class JreProvisioningTest {
+  private static final String PROJECT_KEY = "team-lang-dotnet_jre-provisioning";
+  private static final String PROJECT_NAME = "JreProvisioning";
+
+  @BeforeEach
+  public void setUp() {
+    TestUtils.reset(ORCHESTRATOR);
+  }
+
+  @TempDir
+  public Path basePath;
+
+  @Test
+  void jreProvisioning_skipProvisioning_doesNotDownloadJre() throws Exception {
+    ORCHESTRATOR.getServer().provisionProject(PROJECT_KEY, PROJECT_NAME);
+    String token = TestUtils.getNewToken(ORCHESTRATOR);
+    Path projectDir = TestUtils.projectDir(basePath, PROJECT_NAME);
+
+    var result = ORCHESTRATOR.executeBuild(TestUtils.newScanner(ORCHESTRATOR, projectDir, token)
+      .addArgument("begin")
+      .setProjectKey(PROJECT_KEY)
+      .setProjectName(PROJECT_NAME)
+      .setProperty("sonar.projectBaseDir", Paths.get(projectDir.toAbsolutePath().toString(), PROJECT_NAME).toString())
+      .setProperty("sonar.scanner.skipJreProvisioning", "true")
+      .setProperty("sonar.verbose", "true"));
+    assertThat(result.isSuccess()).isTrue();
+
+    var logs = result.getLogs();
+    assertThat(logs).contains(
+      "JreResolver: Resolving JRE path.",
+      "JreResolver: sonar.scanner.skipJreProvisioning is set, skipping JRE provisioning.");
+    assertThat(logs).doesNotContain(
+      "JreResolver: Cache miss.",
+      "JreResolver: Cache hit",
+      "JreResolver: Cache failure.");
+  }
+
+  @Test
+  void jreProvisioning_endToEnd_cacheMiss_downloadsJre() throws IOException {
+    // provisioning does not exist before 10.6
+    if (!ORCHESTRATOR.getServer().version().isGreaterThanOrEquals(10, 6)) {
+      return;
+    }
```

Comment:
The SQ tests run for a lot of versions, going back until 8.9.
I prefer to do this cheap check here, instead of messing with the  regex of azure pipelines yml.

If you have any other ideas, let me know.

Reply:
This is okay in principal and was done before. The correct way to do it is using [assumeTrue](https://junit.org/junit4/javadoc/latest/org/junit/Assume.html), though:
https://github.com/SonarSource/sonar-scanner-msbuild/blob/8553b2a79819aea5712d949ac10ccfcea27290ef/its/src/test/java/com/sonar/it/scanner/msbuild/sonarqube/ScannerMSBuildTest.java#L940